src/main/java/com/exasol/udfstartuptimeimprover/UdfDefinitionParser.java

```diff
@@ -0,0 +1,116 @@
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-USTI-1")
+                    .message(FAILED_TO_PARSE_SCRIPT_DEFINITION + "The script definition must start with 'CREATE'.")
+                    .toString());
+        }
+    }
+
+    private UdfDefinition readDefinition(final String udfDefinition) {
+        final int indexOfScriptDefinitionStart = getIndexOfScriptDefinitionStart(udfDefinition);
+        final String createStatement = udfDefinition.substring(0, indexOfScriptDefinitionStart);
+        final String scriptContent = udfDefinition.substring(indexOfScriptDefinitionStart);
+        String scriptClass = null;
+        final List<String> jars = new ArrayList<>();
+        final List<String> jvmOptions = new ArrayList<>();
+        for (final String row : scriptContent.split("\n")) {
+            final String trimmedRow = row.trim();
+            if (trimmedRow.startsWith(KEY_SCRIPTCLASS)) {
+                assertScriptClassIsNotAlreadySet(scriptClass);
+                scriptClass = readOption(trimmedRow, KEY_SCRIPTCLASS);
+            } else if (trimmedRow.startsWith(KEY_JAR)) {
+                jars.add(readOption(trimmedRow, KEY_JAR));
+            } else if (trimmedRow.contains(KEY_JVM_OPTION)) {
+                jvmOptions.addAll(Arrays.asList(readOption(trimmedRow, KEY_JVM_OPTION).split("\\s")));
+            } else if (!trimmedRow.isBlank()) {
+                throw getUnexpectedLineException(trimmedRow);
+            }
+        }
+        return new UdfDefinition(createStatement, scriptClass, jars, jvmOptions);
+    }
+
+    private void assertScriptClassIsNotAlreadySet(final String scriptClass) {
+        if (scriptClass != null) {
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-USTI-2")
+                    .message(FAILED_TO_PARSE_SCRIPT_DEFINITION + "'%scriptclass' must only be set once.").toString());
+        }
+    }
+
+    private IllegalArgumentException getUnexpectedLineException(final String trimmedRow) {
+        return new IllegalArgumentException(ExaError.messageBuilder("E-USTI-4")
+                .message(FAILED_TO_PARSE_SCRIPT_DEFINITION + "Unexpected line {{line}}.", trimmedRow)
+                .mitigation("Probably this script uses inline code. Optimizing those scripts is not supported.")
+                .toString());
+    }
+
+    private void assertJarsNotEmpty(final List<String> jars) {
+        if (jars.isEmpty()) {
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-USTI-7")
+                    .message(FAILED_TO_PARSE_SCRIPT_DEFINITION + "Could not find a '%jar' definition.").toString());
+        }
+    }
+
+    private void assertScriptClassIsSet(final String scriptClass) {
+        if (scriptClass == null) {
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-USTI-5")
+                    .message(FAILED_TO_PARSE_SCRIPT_DEFINITION + "Could not find required '%scriptclass'.").toString());
+        }
```

Comment:
Is it necessary to check that the string is not empty?

Reply:
makes sense